keanu-project/src/main/java/io/improbable/keanu/tensor/Tensor.java

```diff
@@ -87,8 +87,20 @@ default boolean isScalar() {
         return getLength() == 1;
     }
 
+    /**
+     * Returns true if the tensor is a vector. A vector being a 1xn or a nx1 tensor.
+     *
+     * (1, 2, 3) is a 1x3 vector.
+     *
+     * (1)
+     * (2)
+     * (3) is a 3x1 vector.
+     *
+     *
+     * @return true if the tensor is a vector
+     */
     default boolean isVector() {
-        return getRank() == 1;
```

Comment:
Having a rank of 1 is also a vector. Even if we don’t actually support rank 1 at the moment.

Reply:
As discussed, I've added the rank one check back in